client.go

```diff
@@ -485,36 +531,12 @@ func (c *TunaSessionClient) DialWithConfig(remoteAddr string, config *nkn.DialCo
 		return nil, err
 	}
 
-	ctx := context.Background()
-	var cancel context.CancelFunc
-	if config.DialTimeout > 0 {
-		ctx, cancel = context.WithTimeout(ctx, time.Duration(config.DialTimeout)*time.Millisecond)
-		defer cancel()
-	}
-
-	buf, err := json.Marshal(&Request{Action: "getPubAddr"})
-	if err != nil {
-		return nil, err
-	}
-
-	respChan, err := c.multiClient.Send(nkn.NewStringArray(remoteAddr), buf, nil)
-	if err != nil {
-		return nil, err
-	}
-
-	var msg *nkn.Message
-	select {
-	case msg = <-respChan.C:
-	case <-ctx.Done():
-		return nil, ctx.Err()
-	}
-
-	pubAddrs := &PubAddrs{}
-	err = json.Unmarshal(msg.Data, pubAddrs)
+	pubAddrs, err := c.getPubAddrsFromRemote(remoteAddr, config)
 	if err != nil {
 		return nil, err
 	}
 
+	ctx := context.Background()
```

Comment:
Timeout is ignored here. You should keep the previous context logic and pass it to `getPubAddrsFromRemote()` as argument